pageserver/src/layered_repository.rs

```diff
@@ -1981,19 +2035,20 @@ impl LayeredTimeline {
         //
         // Compaction creates multiple image layers. It would be better to create them all
         // and fsync them all in parallel.
-        par_fsync::par_fsync(&[
-            image_layer.path(),
-            self.conf.timeline_path(&self.timeline_id, &self.tenant_id),
-        ])?;
+        let mut all_paths = Vec::from_iter(layer_paths_to_upload.clone());
+        all_paths.push(self.conf.timeline_path(&self.timeline_id, &self.tenant_id));
+        par_fsync::par_fsync(&all_paths)?;
 
         // FIXME: Do we need to do something to upload it to remote storage here?
```

Comment:
This fixme seems obsolete. We schedule layers upload right after `create_image_layers()` call